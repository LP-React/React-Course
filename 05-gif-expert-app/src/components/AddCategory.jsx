import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [InputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (InputValue.trim().length <= 1) return;
        onNewCategory(InputValue.trim())
        setInputValue('');
    }

    return (
        <form onSubmit={(event) => {
            onSubmit(event)
        }} aria-label='form'>
            <input type="text" placeholder='Search Gifs' value={InputValue} onChange={(event) => onInputChange(event)} />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,

}
