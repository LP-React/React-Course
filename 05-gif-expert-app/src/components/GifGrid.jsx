import React from 'react';
import {getGif} from '../helpers/getGif'


export const GifGrid = ({ category }) => {


    getGif(category);

    return (
        <>
            <h3>{category}</h3>
            {
                // gifs.map(gif => (
                //     <p>{gif}</p>
                // ))
            }
        </>
    )
}
