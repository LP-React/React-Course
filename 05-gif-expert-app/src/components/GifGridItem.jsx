import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url, id }) => {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p key={id}>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}