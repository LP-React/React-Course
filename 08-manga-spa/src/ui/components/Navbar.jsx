import { Link, NavLink, useNavigate } from 'react-router-dom';

export const Navbar = () => {

    const navigate = useNavigate();

    const onLogout = () => {
        navigate('/login', {
            replace: true
        })
    }


    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-3">

            <Link
                className="navbar-brand"
                to="/"
            >
                Mangas
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink
                        className={({ isActive }) => `nav-item nav-link ${isActive ? 'active' : ''}`}
                        to="/Oyasumi-Punpun"
                    >
                        Oyasumi Punpun
                    </NavLink>

                    <NavLink
                        className={({ isActive }) => `nav-item nav-link ${isActive ? 'active' : ''}`}
                        to="/Dead-Demon-De-De-De-Destruction"
                    >
                        Dead Demon De-De-De-Destruction
                    </NavLink>

                    <NavLink
                        className={({ isActive }) => `nav-item nav-link ${isActive ? 'active' : ''}`}
                        to="/Evangelion"
                    >
                        Evangelion
                    </NavLink>

                    <NavLink
                        className={({ isActive }) => `nav-item nav-link ${isActive ? 'active' : ''}`}
                        to="/Search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                <ul className="navbar-nav ml-auto">

                    <span className='nav-item nav-link text-info'>
                        iL4y
                    </span>

                    <button className='nav-item nav-link btn' onClick={onLogout}>
                        Logout
                    </button>

                </ul>
            </div>
        </nav>
    )
}