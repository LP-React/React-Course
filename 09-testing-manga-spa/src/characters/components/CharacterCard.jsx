import { Link } from 'react-router-dom'

export const CharacterCard = ({ character }) => {

    const characterURL = `/assets/characters/${character.id}.webp`

    return (
        <div className="col">
            <div className="card">
                <div className="row no-gutters">
                    <div className="col-4">
                        <img src={characterURL} alt={character.name} className="card-img" />
                    </div>
                    <div className="col-8">
                        <div className="card-body">
                            <h5>{character.name}</h5>
                            <p className="card-text">{character.manga}</p>
                            <Link to={`/Character/${character.id}`}>Read more...</Link>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
