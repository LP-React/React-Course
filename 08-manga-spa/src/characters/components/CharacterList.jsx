import { useMemo } from 'react'

import { CharacterCard } from './CharacterCard'

import { getCharacterByPublisher } from '../helpers'

export const CharacterList = ({ manga }) => {

    const characters = useMemo(() => getCharacterByPublisher(manga), [manga])


    return (
        <div className='row row-cols-1 row-cols-md-3 g-3'>
            {characters.map(character =>
                <div key={character.id}>
                    <CharacterCard character={character} />
                </div>
            )}
        </div>
    )
}
