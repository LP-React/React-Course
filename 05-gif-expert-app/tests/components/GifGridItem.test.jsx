import { GifGridItem } from "../../src/components/GifGridItem";
import { render, screen } from '@testing-library/react';

describe('Prueba <GifGridItem/>', () => {
    const title = 'Yoasobi';
    const url = 'https://yoasobi.com/';

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifGridItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifGridItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
        expect(screen.getByText(title).innerHTML).toBe(title);
    })
})