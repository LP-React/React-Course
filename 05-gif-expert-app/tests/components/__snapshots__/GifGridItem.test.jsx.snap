// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifGridItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Yoasobi"
      src="https://yoasobi.com/"
    />
    <p>
      Yoasobi
    </p>
  </div>
</div>
`;
